boards/arm/b_u585i_iot02a/b_u585i_iot02a-common.dtsi

```diff
@@ -146,3 +153,6 @@ zephyr_udc0: &usbotg_fs {
 &iwdg {
 	status = "okay";
 };
+
+mikrobus_serial: &usart3 {
+};
```

Comment:
You can't do this. Unless I've missed it, there's no actual mikrobus connector on this board

Reply:
You're right. Ideally there should be a `arduino_usart` definition for this board, in addition to the existing `arduino_i2c` and `arduino_spi`, and we assign the module to that.

There are dedicate UART pins on the Uno header, but it doesn't seem very common in Zephyr to configure them. This is all I could find:

```bash
boards/arm/b_l072z_lrwan1/b_l072z_lrwan1.dts
97:arduino_usart: &usart1 {};
```

Reply:
@erwango Should we add `arduino_usart` in here: https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arm/b_u585i_iot02a/arduino_r3_connector.dtsi

Reply:
Yes, it would work. It's not common because not frequently needed, but it's valid.

Reply:
What will not work is to build mikroe shield with this board though.
Building with `-DSHIELD=mikroe_wifi_bt_click` will get `mikroe_wifi_bt_click.overlay` in dts, which can't build with a board not defining `mikrobus_serial`.
Unfortunately device tree connectors description is too much close to hw reality: if not fitting, it doesn't compile. And to get it work you need to use wires on hw and hack dts...
But unfortunately, I don't have solution to overcome this with a solution acceptable upstream

Reply:
Ok, I forgot about the possibility to add other overlays. That would work indeed.